src/wp-includes/block-template-utils.php

```diff
@@ -519,11 +519,19 @@ function _remove_theme_attribute_from_template_part_block( &$block ) {
  * @return WP_Block_Template Template.
  */
 function _build_block_template_result_from_file( $template_file, $template_type ) {
-	$default_template_types = get_default_block_template_types();
-	$template_content       = file_get_contents( $template_file['path'] );
-	$theme                  = get_stylesheet();
+	$template  = new WP_Block_Template();
+	$cache_key = md5( serialize( $template_file ) );
```

Comment:
$template_file is array, with contains theme, slug and path. Making it a good cache key.